server/events/yaml/manifest_parser.go

```diff
@@ -0,0 +1,117 @@
+package yaml
+
+import (
+	"os"
+	"path/filepath"
+
+	version "github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events/yaml/raw"
+	"github.com/runatlantis/atlantis/server/events/yaml/valid"
+
+	"github.com/lyft/datum/pkg/manifest"
+)
+
+// ManifestYAMLFilename is the name of the config file for each repo.
+const ManifestYAMLFilename = "manifest.yaml"
+
+// ManifestParser find and converts all manifest files that contains
+// orchestration.roots
+type ManifestParser struct{}
+
+type manifestRoot struct {
+	dirPath  string
+	manifest *manifest.Manifest
+}
+
+// ToRepoCfg converts manifestRoots to valid.repoCfg configuration object.
+func (m *ManifestParser) ToRepoCfg(manifests []*manifestRoot) (valid.RepoCfg, error) {
+	var repoCfg valid.RepoCfg
+	repoCfg.Version = 3
+
+	for _, manifestRoot := range manifests {
+		roots := manifestRoot.manifest.Orchestration.Roots.List()
+		for _, root := range roots {
+			project := valid.Project{}
+			projectName := manifestRoot.manifest.Name + "-" + root.Name
```

Comment:
I don't see why we shouldn't just use the root name as the project name

Reply:
The problem is project names need to be unique across atlantis.yaml. If you have monorepo it will be hard to enforce that.

Reply:
this code is analogous to TFE workspace names where we use service name + environment as a name for the workspace.

Reply:
why do project names need to be unique across atlantis.yaml?

Reply:
Doesn't atlantis basically partition everything by repo name?

Reply:
With regards to monorepo with multiple manifests, we can enforce that pretty easily in the code i would imagine.

Reply:
> why do project names need to be unique across atlantis.yaml?
Within a repo project names need to be unique. [Example PR](https://github.com/msarvar/atlantis-local-dev/pull/7/files) has project1 and project2 services with 2 manifest.yamls and both have production and staging root names. When I generate atlantis.yaml I have to put into the root of the repository so I create 1 files with 4 roots 2 for each service. 
It will look like this with root names
```
projects:
  - name: production
     dir: projects1/ops/terraform/production
  - name: staging
     dir: projects1/ops/terraform/staging
  - name: production
     dir: projects2/ops/terraform/production
  - name: staging
     dir: projects2/ops/terraform/staging
```
And atlantis complains about non unique project names. Here is the [code](https://github.com/lyft/atlantis/pull/6/files#diff-41e00fb035828d02ffbcf6208a0d03a6R9) that does the validation. 
But reading the docs it seems to imply that project names are not requires if dirs are separate it is only required if we are using the same dir. I can experiment with it and not include the project name into generated atlantis.yaml. When I tried that in this PR atlantis was throwing an error about missing project name.